### PR TITLE
Add 3.7 features to python wrapper

### PR DESCRIPTION
**Pull request by syurkevi**
Adds a number of new features to the python wrapper:
- af_pad
- meanvar 
- inverse square root
- pinverse to python wrapper
- NN convolve and gradient functions to wrapper
- reduce by key
- confidenceCC

### CODE REVIEW

File: arrayfire/algorithm.py [hunk 1]
```diff
@@ -44,6 +44,33 @@ def _nan_reduce_all(a, c_func, nan_val):
...
+        if d > 1:
+            fnsd = i
+            break
+    return int(fnsd)
+
+def _rbk_dim(keys, vals, dim, c_func):
+    keys_out = Array()
+    vals_out = Array()
+    rdim = _FNSD(dim, vals.dims())
+    print(rdim)
```
Comment (9prady9) [inline arrayfire/algorithm.py]: debug print ?

File: arrayfire/algorithm.py [hunk 2]
```diff
@@ -44,6 +44,33 @@ def _nan_reduce_all(a, c_func, nan_val):
...
+    safe_call(c_func(c_pointer(keys_out.arr), c_pointer(vals_out.arr), keys.arr, vals.arr, c_int_t(rdim)))
+    return keys_out, vals_out
+
+def _nan_rbk_dim(a, dim, c_func, nan_val):
+    keys_out = Array()
+    vals_out = Array()
+    rdim = _FNSD(dim, vals.dims())
+    print(rdim)
```
Comment (9prady9) [inline arrayfire/algorithm.py]: debug print ?

File: arrayfire/data.py [hunk 3]
```diff
@@ -799,6 +799,57 @@ def replace(lhs, cond, rhs):
     else:
         safe_call(backend.get().af_replace_scalar(lhs.arr, cond.arr, c_double_t(rhs)))
 
+def pad(a, beginPadding, endPadding, padFillType = PAD.ZERO):
+    """
+    Pad an array
+
+    This function will pad an array with the specified border size and tiling scheme
```
Comment (9prady9) [inline arrayfire/data.py]: no tiling, just padding along edges of all dimensions.

File: arrayfire/image.py [hunk 4]
```diff
@@ -711,6 +711,49 @@ def regions(image, conn = CONNECTIVITY.FOUR, out_type = Dtype.f32):
                                        conn.value, out_type.value))
     return output
 
+def confidenceCC(image, seedx, seedy, radius, multiplier, iters, segmented_value):
+    """
+    Find the confidence connected components in the image.
```
Comment (9prady9) [inline arrayfire/image.py]: Perhaps a link to our C++ documentation that shows an example set of images. You can add them here too but you would to copy those images to this repo again.

File: arrayfire/lapack.py [hunk 5]
```diff
@@ -264,6 +264,39 @@ def inverse(A, options=MATPROP.NONE):
...
+    Returns
+    -------
+
+    AI: af.Array
+       - A 2 dimensional array that is the pseudo-inverse of `A`
+
+    Note
+    ----
+
+    This function is not supported in GFOR
```
Comment (9prady9) [inline arrayfire/lapack.py]: :+1:

File: arrayfire/image.py (hunk 4 above)
Comment (syurkevi) [reply arrayfire/image.py]: The rest of our python documentation is picture-less. Avoiding those links for now.

**Comment by syurkevi** [comment]
@9prady9 The only missing function I see remaining is the setAxesLabelFormat(). Should this go in with this PR? Can it go separately?

**Comment by 9prady9** [comment]
Not sure, do we need it separately ?

**Pull request merged**